src/coreComponents/physicsSolvers/fluidFlow/kernels/compositional/IHUPhaseFlux.hpp

```diff
@@ -42,7 +42,132 @@ using Deriv = constitutive::multifluid::DerivativeOffset;
 namespace UpwindHelpers
 {
 
-static constexpr double minTotMob = 1e-12;
+template< localIndex numComp >
+GEOS_HOST_DEVICE
+static void assignToZero( real64 & deriv_dP, real64 ( & deriv_dC )[numComp] )
+{
+  deriv_dP = 0.0;
+  for( localIndex ic = 0; ic < numComp; ++ic )
+  {
+    deriv_dC[ic] = 0.0;
+  }
+}
+
+template< localIndex numComp >
+GEOS_HOST_DEVICE
+static void assignToZero( real64 & value, real64 & deriv_dP, real64 ( & deriv_dC )[numComp] )
+{
+  value = 0.0;
+  assignToZero( deriv_dP, deriv_dC );
+}
+
+template< localIndex numComp, localIndex numFluxSupportPoints >
+GEOS_HOST_DEVICE
+static void assignToZero( real64 & value, real64 ( & deriv_dP )[numFluxSupportPoints], real64 ( & deriv_dC )[numFluxSupportPoints][numComp] )
+{
+  value = 0;
+  for( localIndex ke = 0; ke < numFluxSupportPoints; ++ke )
+  {
+    assignToZero( deriv_dP[ke], deriv_dC[ke] );
+  }
+}
+
+template< localIndex numComp >
+GEOS_HOST_DEVICE
+static void addDerivativesScaled( real64 ( &deriv_dP ), real64 ( & deriv_dC )[numComp],
```

Comment:
Great helpers addition though I would def stick to the convention of passing the non-const ref last

Reply:
sure, let me flip those